src/Exception/MultipartUploadException.php

```diff
@@ -55,4 +60,26 @@ public function getState()
     {
         return $this->state;
     }
+
+    /**
+     * Get file path of the transfer
+     *
+     * @return array
+     */
+    public function getFilePaths()
+    {
+        return $this->filePath;
+    }
+
+    /**
+     * Construct file path
+     *
+     * @param CommandInterface $cmd
+     */
+    private function collectPathInfo(CommandInterface $cmd)
+    {
+        if (empty($this->filePath) && isset($cmd['Bucket']) && isset($cmd['Key'])) {
```

Comment:
I don't think this will work... This is now tying this class to S3, but it's also used in Glacier: https://github.com/aws/aws-sdk-php/blob/master/src/Glacier/MultipartUploader.php#L6. Maybe the filePath needs to be injected into the exception in the constructor instead of extracted from a previous exception.